File: src/gasing/division/division1.py
import time

a = 123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890
b = 1

# GASING Division
start = time.time()
for _ in range(10000000):
        hasil = a 
        sisa = 0
end = time.time()
gasing_time = end - start
print("GASING Division 1:", gasing_time, "seconds")
print("GASING Result:", hasil)
print("GASING Remainder:", sisa)

# Computer Division
start = time.time()
for _ in range(10000000):
    hasil = a // b
    sisa = a % b
end = time.time()
computer_time = end - start
print("General Division 1:", computer_time, "seconds")
print("Computer Result:", hasil)
print("Computer Remainder:", sisa)

print("\nGASING is", computer_time/gasing_time, "times faster")

File: src/gasing/division/division1.js
const a = 123456789;
const b = 1;

// GASING Division
let start = performance.now();
let hasil;
let sisa;
for (let i = 0; i < 10000000; i++) {
    hasil = a;
    sisa = 0;
}
let gasing_time = (performance.now() - start) / 1000;
console.log("GASING Division 1:", gasing_time, "seconds");

// Computer Division
start = performance.now();
for (let i = 0; i < 10000000; i++) {
    hasil = Math.floor(a / b);
    sisa = a % b;
}
let computer_time = (performance.now() - start) / 1000;
console.log("General Division 1:", computer_time, "seconds");

console.log("\nGASING is", computer_time/gasing_time, "times faster");